### PR TITLE
Fix inventories

### PR DESCRIPTION
**Pull request by ZephireNZ**
This PR addresses the following issues:
- Crafting Inventory, as well as Enchanting and Anvils, do not show the proper UI to the player. This is fixed by sending a size of 0 - odd, but apparently that's Mojang.
- Many block inventories would no be opened when right-clicking.
- Adds missing title entities for Furnaces, Brewing Stands, etc.

I've done some moderate testing and everything seems to be working fine, but can't hurt to do some more.

Note: this depends on https://github.com/GlowstoneMC/Glowkit/pull/28

### CODE REVIEW

**Comment by gdude2002** [comment]
Build passes in conjunction with https://github.com/GlowstoneMC/Glowkit/pull/28

http://bamboo.gserv.me/browse/GLOW-SRV-PR338-40

**Comment by gdude2002** [comment]
Broken as of the Lombok commit. http://bamboo.gserv.me/browse/GLOW-SRV-PR338-49

File: src/main/java/net/glowstone/block/entity/TEBrewingStand.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import net.glowstone.inventory.GlowBrewerInventory;
+import net.glowstone.util.nbt.CompoundTag;
+
+public class TEBrewingStand extends TEContainer {
+
+    private int brewTime = 0;
+
+    public TEBrewingStand(GlowBlock block) {
+        super(block, new GlowBrewerInventory(new GlowBrewingStand(block, 0)));
+        setSaveId("Cauldron");
```
Comment (turt2live) [inline src/main/java/net/glowstone/block/entity/TEBrewingStand.java:15]: O_o

File: src/main/java/net/glowstone/inventory/GlowEnchantingInventory.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        setItem(ITEM_SLOT, item);
+    }
+
+    @Override
+    public ItemStack getItem() {
+        return getItem(ITEM_SLOT);
+    }
+
+    @Override
+    public void setLapis(ItemStack item) {
```
Comment (turt2live) [inline src/main/java/net/glowstone/inventory/GlowEnchantingInventory.java]: Where did these come from?

Comment (gdude2002) [inline src/main/java/net/glowstone/inventory/GlowEnchantingInventory.java]: https://github.com/GlowstoneMC/Glowkit/pull/28

**Comment by gdude2002** [comment]
Fixed: http://bamboo.gserv.me/browse/GLOW-SRV-PR338-51

File: src/main/java/net/glowstone/block/entity/TEBrewingStand.java (hunk 1 above)
Comment (ZephireNZ) [inline src/main/java/net/glowstone/block/entity/TEBrewingStand.java:15]: It's stupid, but yes that's how MC stores Brewing Stands: http://minecraft.gamepedia.com/Chunk_format#Tile_entity_format

**Comment by gdude2002** [comment]
Broken: http://bamboo.gserv.me/browse/GLOW-SRV-PR338-57

**Comment by ZephireNZ** [comment]
You didn't update to the latest changes in my Glowkit PR, that's why it's breaking.

**Comment by SpaceManiac** [comment]
I like it! Just a couple of nitpicks:
- What's `GlowBlockProjectileSource` for? Why not just reuse the `GlowDispenser`?
- `getNumSlots` is easy to confuse with `getSize`. Maybe something along the lines of `getRawSlotCount`? Kind of verbose though.

**Comment by gdude2002** [comment]
@ZephireNZ That was actually the first thing I checked, lol. I'll rebuild it again, in case you've added commits since I posted my response.

**Comment by gdude2002** [comment]
Passed: http://bamboo.gserv.me/browse/GLOW-SRV-PR338-59

Next time, don't edit your comments. Make new ones.

**Comment by turt2live** [comment]
@ZephireNZ Your PR is non-functional due to it breaking worlds. I didn't test the inventories further after my world was broke. 

What it should look like: 
image

What it looked like: 
image

If you can, please provide insight. (by the way, the wool represents how the inventories function on the current master branch. Yellow denotes 'mostly works')

**Comment by turt2live** [comment]
Talked about this on IRC, here's the summary:
- Above claim of corrupted worlds is false, this is due to a larger issue out of scope of this PR
- Animations of chests do not work
- Double chests do not work (handled by #351)
- Functionality (brewing, enchanting, etc) does not work (out of scope)

Asked to fix animations and then this can be pulled :)

**Comment by turt2live** [comment]
~~@ZephireNZ Your PR does not compile! Here's some information about the compile testing:~~
~~\- Glowkit PR found: false~~
~~\- Glowkit PR parsed: https://github.com/GlowstoneMC/Glowkit/pull/28~~

~~http://bamboo.gserv.me/browse/GLOW-SRV-PR338-40~~
~~\- Glowkit compiled: false~~

~~If your PR requires Glowkit, please ensure that an accurate link to the Glowkit PR appears in the description of your PR. If the information above is incorrect, please contact me on IRC (`turt2live` in `#glowstone` on esper.net) for more information.~~

~~If your PR does not require Glowkit, please fix your PR.~~

~~Thank you!~~

No longer relevant (parse error)

**Comment by ZephireNZ** [comment]
@turt2live Can you check out that build log? Pretty sure it's not me, judging by the error message `Could not find method mavenLocal() for arguments [{artifactUrls=[file:///home/bamboo/maven/repos/glowkit/28/]}] on repository container`

**Comment by turt2live** [comment]
@ZephireNZ from IRC:

```
<turt2live> ZephireNZ, sorry. It's a parse failure on my end. I'm just going to edit your PR description to fix it
<turt2live> Oh, you did
<turt2live> I should really just edit my bot's comment
```

**Comment by turt2live** [comment]
Edited: BOT FAIL (resolved)

**Comment by gdude2002** [comment]
Doesn't it? http://bamboo.gserv.me/browse/GLOW-SRV-PR338-78

**Comment by turt2live** [comment]
I have no idea why it is parsing that instead of the one in the description...

EDIT: FIXED. The link in the description referenced the visible text and did not link. Changed it to a link and it should be okay now.

**Comment by SpaceManiac** [comment]
> Asked to fix animations and then this can be pulled :)

As I mentioned on IRC, if this is delaying things let me know and I am willing to pull as-is. Animations are secondary to the actual functionality.

**Comment by gdude2002** [comment]
> Asked to fix animations and then this can be pulled :)

Open?
EEEERClosed.

EDIT: Bored and referencing the April Fools sounds, if people don't get it. 388 doesn't merge. http://bamboo.gserv.me/browse/GLOW-SRV-PR338-85

EDIT 2: Yes, I know the slug is 338, I typo'd :P

**Comment by ZephireNZ** [comment]
@SpaceManiac I have an idea of how to do it, but I think I'll leave it like this for now. Pull when ready, cap'n :)

**Comment by ZephireNZ** [comment]
Huzzah, it builds: http://bamboo.gserv.me/browse/GLOW-SRV-PR338-85

**Comment by turt2live** [comment]
Only noted issues are double chests (previously discussed), beacons (out of scope of this PR), and trapped chests (just screwed up - out of scope).

Marking as ready for pull.

**Comment by SpaceManiac** [comment]
I notice you add several new `BlockState` implementations, but don't use all of them in the corresponding `TileEntity` classes, so it seems like they won't be used when calling `GlowBlock#getState()`. Can you confirm?

**Comment by turt2live** [comment]
Pulled via https://github.com/GlowstoneMC/Glowstone/commit/80da285c4d2eddfaf5d407aeedb05dce3240b1a8

**Pull request closed without merging**